sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLCache.scala

```diff
@@ -47,8 +76,18 @@ private[connect] class MLCache extends Logging {
    *   the key
    */
   def register(obj: Object): String = {
+    val (estimatedSize, name) = obj match {
+      case m: Model[_] => (m.estimatedSize, m.uid)
+      case o => (SizeEstimator.estimate(o), Option(o).map(_.getClass.getName).getOrElse("NULL"))
+    }
+
+    val maxSize = conf.getConf(Connect.CONNECT_SESSION_ML_CACHE_SINGLE_ITEM_SIZE)
+    if (maxSize > 0 && estimatedSize > maxSize) {
+      throw MlItemSizeExceededException("cache", name, estimatedSize, maxSize)
```

Comment:
Do we need to remove the existing cache before throwing exception?

Reply:
I think we should not stop the whole sessionholder here.
Just throwing an error seems a right behavior